packages/core-kernel/src/providers/plugin-discoverer.ts

```diff
@@ -0,0 +1,68 @@
+import { existsSync, readJSONSync } from "fs-extra";
+import glob from "glob";
+import { join } from "path";
+
+import { Application, Plugin } from "../contracts/kernel/application";
+import { PluginCannotBeFound } from "../exceptions/plugins";
+import { Identifiers, inject, injectable } from "../ioc";
+
+@injectable()
+export class PluginDiscoverer {
+    @inject(Identifiers.Application)
+    private readonly app!: Application;
+
+    private plugins: Plugin[] = [];
+
+    public async initialize(): Promise<void> {
+        this.plugins = [];
+
+        const discoverOnPath = async (patter: string, path: string): Promise<void> => {
+            if (existsSync(path)) {
+                this.plugins = this.plugins.concat(await this.discover(patter, path));
+            }
+        };
+
+        await discoverOnPath("*/package.json", join(__dirname, "../../../../packages")); // Project packages
+        await discoverOnPath("*/package.json", join(__dirname, "../../../../plugins")); // Project plugins
+        await discoverOnPath("*/package.json", this.app.dataPath("plugins")); // Installed plugins
+    }
```

Comment:
Just make a loop.

```typescript
const paths = [/* */];

for (const path of paths) {
  if (existsSync(path)) {
    this.plugins = this.plugins.concat(await this.discover("*/package.json", path));
  }
}
```

Reply:
There is no need to loop. Last discover uses different pattern. Fixed in: Fix packages pattern